### PR TITLE
Support for AEC/CCSDS Compression (DRT 5.42)

### PR DESCRIPTION
**Pull request by EricEngle-NOAA**
Pull request for issue #424.

Changes in this pull request provide support for AEC/CCSDS compression in the g2c library as specified by DRT 5.42.

Tests have been created (tests/tst_aec.c) and are successful on my Fedora 38 VM (gcc 13.2)

### CODE REVIEW

**Review by edwardhartnett — APPROVED** [review]
Nice! Do you want me to merge this in before the 1.8.0 releaase?

File: src/decenc_aec.c [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ * @return
+ * - >0 Length of data from AEC decoder
+ * - 0 Successful decode (AEC_OK)
+ * - -1 AEC_CONF_ERROR
+ * - -2 AEC_STREAM_ERROR
+ * - -3 AEC_DATA_ERROR
+ * - -4 AEC_MEM_ERROR
+ * - -5 AEC_RSI_OFFSETS_ERROR
+ *
+ * @author Eric Engle @data 2023-10-16
```
Comment (edwardhartnett) [inline src/decenc_aec.c]: Should be "date" not "data", this is what is causing your developer build to fail in CI...

File: src/decenc_aec.c [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+ * @return
+ * - >0 Exact length of AEC encoded data.
+ * - 0 Successful decode (AEC_OK)
+ * - -1 AEC_CONF_ERROR
+ * - -2 AEC_STREAM_ERROR
+ * - -3 AEC_DATA_ERROR
+ * - -4 AEC_MEM_ERROR
+ * - -5 AEC_RSI_OFFSETS_ERROR
+ *
+ * @author Eric Engle @data 2023-10-16
```
Comment (edwardhartnett) [inline src/decenc_aec.c]: Should be "date" not "data".

File: src/decenc_aec.c (hunk 1 above)
Comment (EricEngle-NOAA) [reply src/decenc_aec.c]: Fixed with [38bef85](https://github.com/NOAA-EMC/NCEPLIBS-g2c/pull/449/commits/38bef85423c039e7572e565e71154bd0953cf562)

File: src/decenc_aec.c (hunk 2 above)
Comment (EricEngle-NOAA) [reply src/decenc_aec.c]: Fixed with [38bef85](https://github.com/NOAA-EMC/NCEPLIBS-g2c/pull/449/commits/38bef85423c039e7572e565e71154bd0953cf562)

**Comment by EricEngle-NOAA** [comment]
> Nice! Do you want me to merge this in before the 1.8.0 releaase?

@edwardhartnett - When are you planning to release v1.8.0?

**Comment by edwardhartnett** [comment]
I'm planning on releasing 1.8.0 today or early next week...

**Comment by EricEngle-NOAA** [comment]
There were a couple of failed tests that I am working through.  Does not seem major -- type mismatches.  Going to push fixes soon.

**Comment by edwardhartnett** [comment]
This is happening on MacOS:

/Users/runner/work/NCEPLIBS-g2c/NCEPLIBS-g2c/g2c/tests/tst_aec.c:9:10: fatal error: 'libaec.h' file not found
#include <libaec.h>

Do you know how to install the AEC library on MacOS?

**Comment by EricEngle-NOAA** [comment]
Yes.  I am working on the missing aec header for macOS now.

I think it will be best to release early next week if the desire is to get AEC support in v1.8.0

**Comment by EricEngle-NOAA** [comment]
Yay the macOS tests are passing with commit 16499e3

Now onto the Linux options tests

**Comment by edwardhartnett** [comment]
Next week is fine. I'm not ready yet... ;-)

**Comment by edwardhartnett** [comment]
Also do you have any real-world examples of data files compressed with AEC?

**Comment by edwardhartnett** [comment]
What is happening with the CI here? How come nothing is running?

**Comment by EricEngle-NOAA** [comment]
Hmmm... I did put `[skip ci]` in the last commit message for .gitignore.

**Comment by EricEngle-NOAA** [comment]
These 8 jobs do not have aec parameters, so something got messed up somewhere.  Still looking...

**Comment by EricEngle-NOAA** [comment]
@edwardhartnett - I think somewhere in the GitHub CI, these 8 checks are "required", but these test do not exist anymore since I modified Linux_options and MacOS to include AEC parameters.  In other words, these tests don't exist anymore, but they are required.

**Comment by edwardhartnett** [comment]
OK, I forgot that you modified the tests. Stand by...

**Comment by EricEngle-NOAA** [comment]
🎉  Looks good! 🎉

**Comment by edwardhartnett** [comment]
@EricEngle-NOAA I'm going to merge this. Thanks very much for this contribution, it will benefit many other users as well.

Could you please find some smallish real-world data file that uses this kind of packing, so we can have a test file on our FTP site?

**Comment by EricEngle-NOAA** [comment]
@edwardhartnett - It's been hard to find, but I can generate it.  How many messages and what size limit should I consider?

**Comment by edwardhartnett** [comment]
Some small file would be good, so just a couple of messages...

**Pull request merged**